drivers/ad7746/Makefile.include

```diff
@@ -0,0 +1 @@
+USEMODULE_INCLUDES += $(RIOTBASE)/drivers/ad7746/include
```

Comment:
```suggestion
USEMODULE_INCLUDES := $(USEMODULE_INCLUDES) $(LAST_MAKEFILEDIR)/include
```

would that work too?

Reply:
I'm not sure. I think USEMODULE_INCLUDES are appended and includes are the same for all modules (which is a bit crap).

Reply:
> rename / move the drivers around without having to touch Makefile.include

For the rename, I don't see a real benefit since you would still have to rename all API functions, defines, etc. For the move, it's true but are we close to a move of drivers away from `$(RIOTBASE)/drivers` ?

Reply:
> are we close to a move of drivers away from `$(RIOTBASE)/drivers`

I'm still dreaming of a more structured directory layout with `drivers/net`, `drivers/sensors` and so on…

Reply:
I tested your proposal and it works.

Reply:
I would prefer to do it the same way it's done for external modules:

https://github.com/RIOT-OS/RIOT/blob/ef4835a0927bff694753435b8fbdc425c0125b6c/tests/external_module_dirs/external_module/Makefile.include#L2-L3

Only `LAST_MAKEFILEDIR` needs to be evaluated immediately, not `USEMODULE_INCLUDES`.

Reply:
finally, this was easy to adapt:
```
for driver in $(ls); do if [ -f "${driver}/Makefile.include" ]; then printf "USEMODULE_INCLUDES_${driver} := \$(LAST_MAKEFILEDIR)/include\n" > ${driver}/Makefile.include; printf "USEMODULE_INCLUDES += \$(USEMODULE_INCLUDES_${driver})\n" >> ${driver}/Makefile.include; fi; done
```